cupy/core/core.pyx

```diff
@@ -351,7 +352,10 @@ cdef class ndarray:
         .. seealso:: :meth:`numpy.ndarray.fill`
 
         """
-        elementwise_copy(value, self, dtype=self.dtype)
+        if value == 0 and self._c_contiguous:
+            self.data.memset_async(0, self.nbytes, stream.Stream(True))
```

Comment:
This memset ignores a sign of the zero. Is it okay? (I think it doesn't matter.)